tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/DockerUtils.java

```diff
@@ -1,15 +1,38 @@
 package org.graalvm.internal.tck;
 
+import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class DockerUtils {
 
     public static Set<String> getAllowedImages() throws IOException {
-        return new HashSet<>(Files.readAllLines(Paths.get("./tests/tck-build-logic/src/main/resources/AllowedDockerImages.txt")));
+        String dockerfileDirectory = Paths.get("./tests/tck-build-logic/src/main/resources/allowed-docker-images").toString();
+        File[] dockerFiles = new File(dockerfileDirectory).listFiles();
+        if (dockerFiles == null) {
+            throw new RuntimeException("Cannot find allowed-docker-images directory content");
+        }
+
+        final String FROM = "FROM";
+        Set<String> allowedImages = new HashSet<>();
+        for (File dockerFile : dockerFiles) {
+            List<String> images = Files.readAllLines(dockerFile.toPath())
+                    .stream()
+                    .filter(line -> line.startsWith(FROM))
```

Comment:
There can only ever be one `FROM` in a Dockerfile, right? At least, we should stop at the first, and fail if there is none. Could you take care of that please?